### PR TITLE
Add completion fields to appliction

### PR DESCRIPTION
**Pull request by colinbruce**
This will allow us to record the user that completes the
application form.

It also aligns the models with similar properties, allowing
us to share code to set the attributes.

Setting the values will be implemented by the up-coming 
ResolverService

This will be needed for the manager dashboard changes at [ticket](https://www.pivotaltracker.com/story/show/107484086)

### CODE REVIEW

**Comment by colinbruce** [comment]
Questions:

* Should we code for completed_by being empty or should we migrate,
    * created_by to completed_by?
    * created_on to completed_at?
* Update query/processed_applications to look at where completed_at NOT nil?

File: db/schema.rb [hunk 1]
```diff
@@ -31,12 +31,6 @@
   add_index "applicants", ["application_id"], name: "index_applicants_on_application_id", using: :btree
 
   create_table "applications", force: :cascade do |t|
-    t.string   "title"
-    t.string   "first_name"
-    t.string   "last_name"
-    t.date     "date_of_birth"
-    t.string   "ni_number"
-    t.boolean  "married"
     t.string   "status"
```
Comment (colinbruce) [inline db/schema.rb]: I've just noticed that these field have been deleted from my schema, despite me not touching them... I'm going to update the file and force push this branch.

**Comment by novotnyjakub** [comment]
So there are no `created_on` and `created_by` fields in the db ;-) But it's a very good point and I think we should try to migrate if we can. 

But I'd make sure that we only migrate the ones which have `application_outcome` set and which have `status` set to `active` (or whichever represents the application has been completed). 

For the applications with that condition matching, I think we can migrate 
* `created_at` -> `completed_at`
* `user_id` -> `completed_by`

I'd even consider to later remove the `user_id` column as is, because it won't have much value.

**Comment by colinbruce** [comment]
Looking at this now and we have an issue... the status column was used (and set) by wicked for the wizard.  Consequently it now doesn't reach, or pass the summary stage.  I'm trying to work out an algorithm now. :(

**Comment by novotnyjakub** [comment]
So my main concern was not to migrate the 'zombie' applications, but maybe we can assume that if it has income and outcome setup, it's `completed`?

**Pull request closed without merging**